packages/service/src/utils/document-history.ts

```diff
@@ -0,0 +1,38 @@
+import * as Y from "yjs";
+import { pool, db } from "../db";
+import gitDiff from "git-diff";
+
+function fetchDocumentHistory(document_id: string) {
+  return db
+    .select(
+      "document_updates_queue",
+      { document_id },
+      {
+        order: { by: "created_at", direction: "ASC" },
+        lateral: {
+          user: db.selectExactlyOne(
+            "users",
+            { id: db.parent("user_id") },
+            { columns: ["id", "name"] }
+          ),
+        },
+      }
+    )
+    .run(pool);
+}
+
+export async function buildDocumentHistory(document_id: string) {
+  const history = await fetchDocumentHistory(document_id);
+  const ydoc = new Y.Doc();
+  return history.reduce((acc, update) => {
```

Comment:
This list gets pretty huge really fast. Processing them on the request thread is a terrible idea.

Since we're already processing document updates in a worker thread, maybe we can keep do this update there also.